pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -209,8 +200,8 @@ func (h *handler) getSecretData(meta metav1.Object, obj data.Object, create bool
 		cloudCredentialSecretName = oldCredential
 	}
 
-	if cloudCredentialSecretName != "" {
-		secret, err := GetCloudCredentialSecret(h.secrets, meta.GetNamespace(), cloudCredentialSecretName)
+	if cloudCredentialSecretName != "" && machine != nil {
```

Comment:
This check is not necessary because we check `machine == nil` above.

Reply:
I thought the same but Goland is yelling about this. It's due to the `machine == nil ` being checked alongside `create`. If we checked them separately it would not be an issue.

The first screenshot is with the `machine == nil` check I added commented out and goland complaining. The second screenshot is with my `machine == nil` check present and the `create` bool commented out above. 

![image](https://user-images.githubusercontent.com/55704795/153955222-cb4db3c8-4bcd-487a-ab42-e40d29b8be01.png)

![image](https://user-images.githubusercontent.com/55704795/153955065-80d48bc5-7b0e-4e54-bdde-d3311234fbd4.png)

Reply:
Ah, you're right. I'm sorry for the confusion.